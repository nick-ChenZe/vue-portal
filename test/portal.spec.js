// there are two problems still existed
// 1. snapshot title eport incorrect
// 2. snapshot not invoke render after executing $forceUpdate in component

import { mount } from 'avoriaz'
import sinon from 'sinon'
import PortalTestComponent from './portal.test.vue'

it('should render portal component correctly', () => {
  const onChildrenMount = sinon.stub()
  const wrapper = mount(PortalTestComponent, {
    attachToDocument: true,
    propsData: { onChildrenMount }
  })
  
  expect(document.body.innerHTML).toMatchSnapshot()
  expect(expect(onChildrenMount).toBeCalled)

  wrapper.destroy()
})

it('should not render portal component correctly', () => {
  const wrapper = mount(PortalTestComponent, {
    attachToDocument: true,
    propsData: { isHide: true }
  })
  expect(document.body.innerHTML).toMatchSnapshot()
  wrapper.destroy()
})
