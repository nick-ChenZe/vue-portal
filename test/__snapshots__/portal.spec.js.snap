// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render portal component correctly 1`] = `

<div class="parent">
</div>
<div role="portal"
     style="display: none;"
     data-custom="any"
     class="classA"
>
  <button>
    content button
  </button>
</div>

`;

exports[`should render portal component correctly 1`] = `

<div class="parent">
</div>

`;
